### PR TITLE
feat: 設定 YAML とコアユーティリティ（config / seed / checkpoint）の追加

### PR DESCRIPTION
**Pull request by koki3070**
## 概要

PR1（リポジトリ基盤）の上に、**設定ファイル**と **`tda_ml` の薄いユーティリティ層**を追加する PR です（**分割 PR の 2 本目**）。

- **設定:** 正本 YAML（`configs/base.yaml` + `reproduce` / `dev` / `prod` / `test_fast`）と `configs/README.md`（マージ規則・`Trainer` が読むキー一覧）。
- **パッケージ:** `tda_ml.config`（`load_config`、リポジトリ root 基準）、`seed_utils`、`checkpoint_io`、`runtime_profile`、`torch_util`、`__init__.py`。
- **テスト:** 上記に対応する 4 本（`test_checkpoint_io` は **ダミー state dict のみ**で検証し、`models` への依存なし）。

**Depends on #52**（PR1）。PR1 マージ前は、この PR の差分に PR1 の変更も含まれます。

## この PR に含めないもの（予定）

- `tda_ml/models.py`、`data_loader.py`、`trainer.py`、`losses.py` など学習本体
- `experiments/run_backend_multiseed.py`
- `configs/archive/`（`.gitignore` 対象・ローカルのみ）
- CI の `pytest` / 再現スモーク（**引き続き ruff のみ**；`tda_ml` 全体と公式 multiseed が揃った後続 PR で追加）

## レビューの見方

- **`configs/README.md`:** 正本 5 ファイルと、現行スタックが実際に読むキーの対応表が正本です。
- **`load_config`:** `project_root` はパッケージ位置から推定（cwd 非依存）。
- **`test_checkpoint_io.py`:** 本 PR では `AnisotropicOutlierClassifier` を使わず、最小テンソル辞書で `load_torch_checkpoint` / `extract_model_state_dict` を検証しています。

## テスト計画

- [ ] CI: `Ruff` ワークフローが成功すること
- [ ] （任意・ローカル）`uv run pytest tests/test_config_project_root.py tests/test_seed_utils.py tests/test_checkpoint_io.py tests/test_runtime_profile.py`

### CODE REVIEW

File: configs/dev.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Lightweight local training (non-official; for wiring checks).
+
+meta:
+  config_id: "development"
```
[the remark this answers is not in the document]
Comment (t-uda) [reply configs/dev.yaml]: 却下

File: tda_ml/config.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    project_root:
+        Root that contains ``configs/base.yaml``. Defaults to the repository root
+        inferred from this package location (not the process current working directory).
+    """
+    root = Path(project_root).resolve() if project_root is not None else default_project_root()
+
+    base_path = root / "configs" / "base.yaml"
+    with base_path.open("r", encoding="utf-8") as f:
+        base_config = yaml.safe_load(f)
+
```
[the remark this answers is not in the document]
Comment (koki3070) [reply tda_ml/config.py:41]: load_config で `yaml.safe_load` の戻り値が dict でない場合は、ファイルパスと型名を含む `ValueError` を出すようにしました。

File: tda_ml/config.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    return dst
+
+
+def model_kwargs_from_config(config: dict[str, Any]) -> dict[str, Any]:
+    """Keyword arguments for :class:`tda_ml.models.AnisotropicOutlierClassifier`."""
+    model_cfg = config.get("model") or {}
+    return {
+        "point_dim": int(model_cfg.get("point_dim", 2)),
+        "feature_dim": int(model_cfg.get("feature_dim", 128)),
+    }
```
[the remark this answers is not in the document]
Comment (koki3070) [reply tda_ml/config.py:71]: ellipse_param_dim はコミット 46c118c で意図的に削除済みです（楕円パラメータは `[a,b,θ]` の 3 要素で、中心は `points` 側）。`model_kwargs_from_config` と `configs/README.md` はその前提に揃えています。

File: tda_ml/checkpoint_io.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    try:
+        return torch.load(path, **common_kw, weights_only=weights_only)
+    except TypeError:
+        # PyTorch without ``weights_only`` keyword.
+        return torch.load(path, **common_kw)
+    except Exception as exc:
+        if not weights_only:
+            raise
+        warnings.warn(
+            f"torch.load(weights_only=True) failed ({type(exc).__name__}: {exc!s}); "
```
[the remark this answers is not in the document]
Comment (koki3070) [reply tda_ml/checkpoint_io.py:58]: weights_only=True が原因のときだけ weights_only=False にフォールバックし、FileNotFoundError などはそのまま再送出するようにしました。

File: tda_ml/checkpoint_io.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            return torch.load(path, **common_kw)
+
+
+def _looks_like_pytorch_state_dict(obj: Any) -> bool:
+    """Heuristic: string keys and at least one tensor value (typical ``nn.Module`` state)."""
+    if not isinstance(obj, dict) or not obj:
+        return False
+    if not all(isinstance(k, str) for k in obj):
+        return False
+    return any(isinstance(v, torch.Tensor) for v in obj.values())
```
[the remark this answers is not in the document]
Comment (koki3070) [reply tda_ml/checkpoint_io.py]: _looks_like_pytorch_state_dict は全値が torch.Tensor のときだけ真になるよう厳格化しました。

File: configs/dev.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (koki3070) [reply configs/dev.yaml]: configs/dev.yaml の config_id を "dev" に変更し、ファイル名と出力ディレクトリ接頭辞を揃えました。

File: configs/base.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  num_outliers: 20
+  num_workers: 2
+  pin_memory: true
+  seed: 42
+  train_size: 4500
+  val_size: 500
+  test_size: 1000
+  dataset_type: "mnist"
+
+device: "auto" # "auto", "cpu", or "cuda"
```
[the remark this answers is not in the document]
Comment (koki3070) [reply configs/base.yaml]: configs/base.yaml の device コメントに mps を追記し、configs/README.md と一致させました。

File: configs/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
+# Configuration layout
+
+Canonical YAML files live **in this directory** (deep-merged with `base.yaml` by `tda_ml.config.load_config`):
```
Comment (t-uda) [inline configs/README.md:3]: Follow-up: `tda_ml.config.load_config` is 何？

**Review by t-uda — APPROVED** [review]
LGTM

**Review by t-uda — APPROVED** [review]
LGTM
